utils/measure/measure/util/measure_util.py

```diff
@@ -61,16 +163,29 @@ def take_measurement(
 
             if measurement:
                 # Check if measurement is not outdated
-                if measurement.updated < start_timestamp:
+                if start_timestamp and measurement.updated < start_timestamp:
                     error = OutdatedMeasurementError(
                         "Power measurement is outdated. Aborting after %d successive retries",
                         self.config.max_retries,
                     )
 
+                power = measurement.power
+
                 # Check if we not have a 0 measurement
-                if measurement.power == 0:
+                if round(power, 2) <= 0:
```

Comment:
Not sure why we need to round here, can you explain?

Reply:
Yeah, type is float, and we also fetch fetch a floating point number with unknown precision and rounding implementations as text. Those numbers could be reported back by some implementations as `0.00000000000001` or `-0.00000000000001`instead of `0.00`. So we round to make sure these artifacts are removed, before we compare it.